app/src/main/java/com/vrem/wifianalyzer/wifi/collectdata/CollectDataFragment.java

```diff
@@ -0,0 +1,142 @@
+                        writeJSONtoFile(MainContext.INSTANCE.getContext(), currentLocationEt.getText().toString(), jsonObject);
+                    } catch (JSONException e) {
+                        e.printStackTrace();
+                    }
+            }
+        });
+
+       /* readJSON(currentLocationEt.toString());*/
+        return view;
+    }
+
+
+    @Override
+    public void update(@NonNull WiFiData wiFiData) {
+
+        Settings settings = MainContext.INSTANCE.getSettings();
+        WiFiBand wiFiBand = settings.getWiFiBand();
+        Predicate<WiFiDetail> predicate = new WiFiBandPredicate(wiFiBand);
+        WiFiData.wiFiDetails = wiFiData.getWiFiDetails(predicate, SortBy.STRENGTH);
+    }
+
+    private String readJSON(String filename) {
+        File directory = Environment.getExternalStorageDirectory();
+        File file = new File(directory + "/Notes/", filename);
+
+        StringBuilder text = new StringBuilder();
+        try {
+            BufferedReader br = new BufferedReader(new FileReader(file));
+            String line;
+
+            while ((line = br.readLine()) != null) {
+
+                text.append(line);
+                text.append('\n');
+            }
+            JSONObject jsonObject = new JSONObject(String.valueOf(text));
+            for (int i = 0; i<jsonObject.length(); i++) {
+                if (!objectString.contains(jsonObject.getString((String) jsonObject.names().get(i)))) {
+                    objectString = objectString + " " + jsonObject.getString((String) jsonObject.names().get(i)) + "\n";
+                }
+            }
+            /* Toast.makeText(mContext, objectString, Toast.LENGTH_SHORT).show();*/
+            br.close();
+        } catch (IOException e) {
+
+        } catch (JSONException e) {
+            e.printStackTrace();
+        }
+        return text.toString();
+    }
+
+    public void writeJSONtoFile(Context context, String sFileName, JSONObject jsonObject) throws JSONException {
+        // this function (writeJSONtoFile) is called every 5 seconds and every time it sends values of sKey(MAC) and sValue(NAME) that I want to store in a flattened JSONObject
+       //sKey is MAC address of the wifi, sValue is the name of the Wifi
+        try {
+            File root = new File(Environment.getExternalStorageDirectory(), "Notes");
+            if (!root.exists()) {
+                root.mkdirs();
+            }
+            File gpxfile = new File(root, sFileName);
+            FileWriter writer = new FileWriter(gpxfile);
+            Log.d(TAG, "writeJSONtoFile");
+            writer.append(jsonObject.toString()+"\n");
+            writer.flush();
+            writer.close();
+            Toast.makeText(MainContext.INSTANCE.getContext(), "Saved", Toast.LENGTH_SHORT).show();
+        } catch (IOException e) {
+            e.printStackTrace();
+            Toast.makeText(context, "Not saved", Toast.LENGTH_SHORT).show();
```

Comment:
seems like your code is not properly layered, both doing I/O and interacting with the frontend. This makes your code hard to reuse and test